### PR TITLE
Allow users to send access request messages via Discord

### PR DESCRIPTION
**Pull request by vabold**

### CODE REVIEW

**Comment by vabold** [comment]
Ideally I'd still like to wait for the logo asset before merge, but it's not a blocker and that can easily be its own commit anyway.

File: srepanel/web/request_access.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	Author AccessRequestEmbedAuthor  `json:"author"`
+	Fields []AccessRequestEmbedField `json:"fields"`
+}
+
+// https://discord.com/developers/docs/resources/webhook#execute-webhook
+type AccessRequestMessage struct {
+	Username  string               `json:"username"`
+	AvatarURL string               `json:"avatar_url"`
+	Embeds    []AccessRequestEmbed `json:"embeds"`
+}
```
Comment (terorie) [inline srepanel/web/request_access.go:40]: These should be moved into a `discord` package and named after the API types, so we can reference them with e.g. `discord.EmbedAuthor`. Alternatively we can import https://github.com/bwmarrin/discordgo

Comment (terorie) [reply srepanel/web/request_access.go:40]: Sounds good to me

Comment (vabold) [reply srepanel/web/request_access.go]: Resolved in 22ccd92.

File: srepanel/common/types.go [hunk 2]
```diff
@@ -1,8 +1,9 @@
 package common
 
 type Identity struct {
-	ID       uint64 `json:"id"`
-	Username string `json:"username"`
+	ID         uint64 `json:"id"`
+	Username   string `json:"username"`
+	AvatarHash string `json:"avatar"`
```
Comment (terorie) [inline srepanel/common/types.go:6]: Looks fine for now. I suggest putting user information in the SQLite database long term, so the JWT doesn't grow without bounds.

File: srepanel/web/request_access.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+		http.Error(wr, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	if err := req.ParseForm(); err != nil {
+		http.Error(wr, "Bad request", http.StatusBadRequest)
+		return
+	}
+
+	// TODO query ACL to ensure user account exists
```
Comment (terorie) [inline srepanel/web/request_access.go:32]: I don't understand this TODO comment ... If the username doesn't exist in the UserManager, we can just add them before authorizing.

**Review by riptl — APPROVED** [review]

**Comment by riptl** [comment]
Thank you for the contribution @vabold 🎉

**Pull request merged**